cts/cts-attrd.in

```diff
@@ -0,0 +1,324 @@
+#!@PYTHON@
+""" Regression tests for Pacemaker's attribute daemon
+"""
+
+# pylint doesn't like the module name "cts-attrd" which is an invalid complaint for this file
+# but probably something we want to continue warning about elsewhere
+# pylint: disable=invalid-name
+# pacemaker imports need to come after we modify sys.path, which pylint will complain about.
+# pylint: disable=wrong-import-position
+
+__copyright__ = "Copyright 2023 the Pacemaker project contributors"
+__license__ = "GNU General Public License version 2 or later (GPLv2+) WITHOUT ANY WARRANTY"
+
+import argparse
+import os
+import subprocess
+import sys
+import tempfile
+
+# These imports allow running from a source checkout after running `make`.
+# Note that while this doesn't necessarily mean it will successfully run tests,
+# but being able to see --help output can be useful.
+if os.path.exists("@abs_top_srcdir@/python"):
+    sys.path.insert(0, "@abs_top_srcdir@/python")
+
+# pylint: disable=comparison-of-constants,comparison-with-itself,condition-evals-to-constant
+if os.path.exists("@abs_top_builddir@/python") and "@abs_top_builddir@" != "@abs_top_srcdir@":
+    sys.path.insert(0, "@abs_top_builddir@/python")
+
+from pacemaker.buildoptions import BuildOptions
+from pacemaker.exitstatus import ExitStatus
+from pacemaker._cts.corosync import Corosync
+from pacemaker._cts.process import killall, exit_if_proc_running
+from pacemaker._cts.test import Test, Tests
+
+TEST_DIR = sys.path[0]
+
+def update_path():
+    """ Set the PATH environment variable appropriately for the tests """
+
+    new_path = os.environ['PATH']
+    if os.path.exists("%s/cts-attrd.in" % TEST_DIR):
+        # pylint: disable=protected-access
+        print("Running tests from the source tree: %s (%s)" % (BuildOptions._BUILD_DIR, TEST_DIR))
+        # For pacemaker-attrd
+        new_path = "%s/daemons/attrd:%s" % (BuildOptions._BUILD_DIR, new_path)
+
+    else:
+        print("Running tests from the install tree: %s (not %s)" % (BuildOptions.DAEMON_DIR, TEST_DIR))
+        # For pacemaker-attrd
+        new_path = "%s:%s" % (BuildOptions.DAEMON_DIR, new_path)
+
+    print('Using PATH="{}"'.format(new_path))
```

Comment:
not in 3.4

Reply:
This is also present in cts-fencing.in and cts-exec.in.  I'll add another patch to remove it from those files.

Reply:
Whoops. CI doesn't run those, and we're probably the only humans who do ...

Reply:
Definitely leave it in -- I meant that's why we never caught it before